src/models/clsAdapter.js

```diff
@@ -23,13 +26,18 @@ class ContextFactory {
 
         return async function (ctx, next) {
             await new Promise(namespace.bind(function (resolve, reject) {
+                const correlationId = ctx.get(correlationIdHeader) || generateRandomAlphanumeric(correlationIdLength);
+                const sessionId = ctx.get(sessionIdHeader) || ctx.get('x-device-id') || ctx.get('deviceid') || null;
```

Comment:
where the "deviceid" header came from?

Reply:
https://storyous.loggly.com/search#terms=json.correlationId:*%20json.appName:%22api-menu%22%20json.correlationId:%221o3nQd%22&from=2020-12-13T04:00:00.000Z&until=2020-12-13T06:00:00.000Z&source_group=&filter=tag;menu-api

```
headersRequest:
accept-encoding:gzip
androidversion:5.0.1
connect-time:2
connection:close
deviceid:5735e37f05f4840e00faf702
host:api-menu-production.herokuapp.com
if-none-match:"c9335498-3882-11eb-b784-0283ef2b9b0e"
merchantid:573305cbec25610e00f8d729
personid:5f037c3ad3facc004314bebf
placeid:5735e36105f4840e00faf6ff
posversion:1423
total-route-time:0
user-agent:okhttp/3.12.10
via:1.1 vegur, 1.1 vegur
x-authorized:true
x-client-id:55e56c318bd766d774212420
x-device-id:5735e37f05f4840e00faf702
x-forwarded-for:46.135.34.157, 3.250.133.46
x-forwarded-port:443
x-forwarded-proto:https
x-merchant-id:573305cbec25610e00f8d729
x-person-id:5f037c3ad3facc004314bebf
x-place-id:5735e36105f4840e00faf6ff
x-request-id:cb56c208-a7b4-401b-85fc-a5139b7b80a0
x-request-start:1607839198960
```